code/modules/mob/living/simple_animal/hostile/retaliate/fish.dm

```diff
@@ -27,4 +27,43 @@
 	speak_emote = list("gnashes")
 
 	faction = list("carp")
-	flying = 1
\ No newline at end of file
+	flying = 1
+
+
+/mob/living/simple_animal/hostile/retaliate/carp/koi
+	name = "space koi"
+	desc = "a gentle space faring koi."
+	icon = 'icons/obj/fish_items.dmi'
+	icon_state = "koi1"
+	icon_living = "koi1"
+	icon_dead = "koi1-dead"
+	icon_gib = "carp_gib"
+
+	harm_intent_damage = 1
+	melee_damage_lower = 2
+	melee_damage_upper = 2
+	attacktext = "bites"
+	attack_sound = 'sound/weapons/bite.ogg'
+	speak_emote = list("blurps")
+	butcher_results = null
+
+	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+	minbodytemp = 0
+	maxbodytemp = 1500
+
+	faction = list("carp")
+	flying = 1
```

Comment:
why isn't this on the parent mob?

Reply:
Sea carp don't fly. They swim in the diving area. Though I guess that is similar to flight...